tests/vcs/VCSCommitBearTest.py

```diff
@@ -0,0 +1,223 @@
+        os.chmod(path, stat.S_IWRITE)
+        func(path)
+
+    def tearDown(self):
+        os.chdir(self._old_cwd)
+        if platform.system() == 'Windows':
+            onerror = self._windows_rmtree_remove_readonly
+        else:
+            onerror = None
+        shutil.rmtree(self.gitdir, onerror=onerror)
+
+    def test_analyze_git_commit(self):
+        run_shell_command('git checkout -b testbranch master')
+        run_shell_command('touch testfile1.txt ')
+        run_shell_command('git add testfile1.txt')
+        run_shell_command('git commit -m "Add testfile1"')
+        test_sha1 = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+        self.assertEqual(self.run_uut(),
+                         [('Add testfile1\n\n', test_sha1, ['merge_commit'],
+                           ['testfile1.txt'], [], [])])
+
+        run_shell_command('touch testfile2.txt')
+        run_shell_command('git add testfile2.txt')
+        run_shell_command('git commit -m "Add testfile2 [ci skip]"')
+        test_sha2 = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+        self.assertEqual(self.run_uut(),
+                         [('Add testfile2 [ci skip]\n\n', test_sha2,
+                           ['ci_skip_commit', 'merge_commit'],
+                           ['testfile2.txt'], [], [])])
+
+        run_shell_command(['echo "hello world" > testfile2.txt'], shell=True)
+        run_shell_command('sed -i "s/hello/hi/" testfile2.txt')
+        run_shell_command('git add testfile2.txt')
+        run_shell_command('git commit -m "Modify testfile2"')
+        test_sha3 = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+        self.assertEqual(self.run_uut(),
+                         [('Modify testfile2\n\n', test_sha3,
+                           ['merge_commit'],
+                           [], ['testfile2.txt'], [])])
+
+        run_shell_command('touch testfile3.txt')
+        GitCommitBearTest.run_git_command('add', 'testfile3.txt')
+        GitCommitBearTest.run_git_command(
+            'commit', '-m', '"another commit [skip ci]"')
+        test_sha3 = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+        GitCommitBearTest.run_git_command('revert', 'HEAD', '--no-edit')
+        test_sha4 = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+        test_raw_commit_msg = ('Revert "another commit [skip ci]"\n\n'
+                               'This reverts commit %s.\n\n' % (test_sha3))
+        self.assertEqual(self.run_uut(),
+                         [(test_raw_commit_msg, test_sha4,
+                           ['ci_skip_commit', 'merge_commit', 'revert commit'],
+                             [], [], ['testfile3.txt'])])
+
+        run_shell_command('git checkout master')
+        run_shell_command('git checkout -b new-feature master')
+        run_shell_command('touch testfile4.txt ')
+        GitCommitBearTest.run_git_command('add', 'testfile4.txt')
```

Comment:
I think it will be better if we start to take a consistency here, I saw you use `run_shell_command('git..)` and `GitCommitBear.run_git_command` back and forth, I think just use `run_shell_command` in all of the test would be alright